mono/utils/mono-threads-coop.h

```diff
@@ -107,4 +107,36 @@ mono_threads_enter_gc_unsafe_region_unbalanced_with_info (THREAD_INFO_TYPE *info
 
 G_END_DECLS
 
+void
+mono_threads_enter_no_safepoints_slow (MonoThreadInfo *info, const char *func);
+
+void
+mono_threads_exit_no_safepoints (MonoThreadInfo *info, const char *func);
+
+static inline gboolean
+mono_threads_enter_no_safepoints (MonoThreadInfo *info, const char *func)
+{
+	gboolean are_safepoints_enabled = mono_threads_are_safepoints_enabled ();
+	if (are_safepoints_enabled)
+		mono_threads_enter_no_safepoints_slow (info, func);
+	return are_safepoints_enabled;
+}
+
+// Use these macro over brief spans of code,
+// that have raw pointers, but have otherwise
+// been converted to be cooperative suspend / precise
+// stack scan correct.
+//
+// Such spans of code will neither initiate a cooperative
+// suspend nor cooperate with such a request from another thread.
+// Therefore extending pause times under memory pressure,
+// or exhausting memory only due to refusal to collect.
+// Therefore such spans of code should be rare and small.
+#define MONO_ENTER_NO_SAFEPOINTS						\
```

Comment:
One reason I had these in `mono-threads-api.h` is because that's where they logically belong along with the other user-facing thread transitions (enter/exit GC Safe/Unsafe).  I think there's a benefit to keeping the entire API in one place.